fix(shared-data): Re-apply update tip overlap values for Flex Pipettes (#15147)

sfoster1 opened the pull request:
This is #15147 again but this time outside the context of an extremely imminent release. 

While this code needs to be merged, we should carefully consider how we do so. We can't ship this until we can separate this behavior by API level so that customers who may have tuned in offsets  based on the old incorrect tip offsets won't have their protocols invalidated - this data needs to be loaded by API level.

Closes EXEC-452

Carlos-fernandez committed:
fix(shared-data) Update tip overlap values for Flex Pipettes (#15147)

After conducting hardware testing, we've formulated updated tip overlap
values for every tip size and pipette model. These adjustments apply to
1-channel, 8-channel, and 96-channel pipettes. These revised values aim
to mitigate overpressure occurrences in ABR.

[https://docs.google.com/spreadsheets/d/16dEmqulO4SIKU6E0aYTfVQEnZRTn3MwNF_WB8K1KmRM/edit#gid=0](url)

- Added v2 tip overlap values for all flex pipette-tip combos

SyntaxColoring reviewed (approved):
Approved previously in #15147, looks good if and when the versioning considerations are solved.

sfoster1 committed:
feat(api): API 2.19 uses updated tip overlap values

Change API version 2.19 to use the new tip overlap values that should
reduce overpressure errors.

Closes EXEC-452

sfoster1 committed:
glad i had this test

sfoster1 committed:
correct data values

Carlos-fernandez reviewed (approved):
All looks correct! Good to go!

DerekMaggio commented on api/tests/opentrons/protocol_api/core/engine/test_overlap_versions.py:
```diff
@@ -0,0 +1,26 @@
+    """Versions above 2.19 should use v1."""
+    assert overlap_for_api_version(api_version) == "v1"
+
+
```
Can you add a test to verify that a bump to the major version is evaluated correctly, please? So something like `APIVersion(3, 12)` 
I can imagine this being something missed because a 3.x would likely be years from now and subsequently breaking everything.

DerekMaggio replied:
Actually I might be incorrect, it looks like the underlying versions_between would throw an assertion error https://github.com/Opentrons/opentrons/blob/3643682c0c213286b8c31666f5aabfefeadab705/api/tests/opentrons/protocol_api/__init__.py#L102C1-L104C75
If that is true, I will change to approval

sfoster1 replied:
Yes, that's true @DerekMaggio

TamarZanzouri commented on api/src/opentrons/protocol_api/core/engine/__init__.py:
```diff
@@ -12,6 +12,7 @@
 ENGINE_CORE_API_VERSION: Final = APIVersion(2, 14)
 SET_OFFSET_RESTORED_API_VERSION: Final = APIVersion(2, 18)
 
+
```
snuck in there

TamarZanzouri reviewed (approved):
Nice! LGTM!

skowalski08 reviewed (approved):
tested 2.18,2.19, and json -lgtm

sfoster1 committed:
oops
